app/support/reservations/duration_change_validations.rb

```diff
@@ -37,6 +36,8 @@ def start_time_not_changed
   end
 
   def duration_not_shortened
+    return unless reservation.reserve_start_at && reservation.reserve_end_at
```

Comment:
You can use `Reservation#has_reserved_times?` here for a slightly-nicer read, if you want